web/client/src/api/index.ts

```diff
@@ -302,6 +304,75 @@ export function useMutationApiSaveFile(
   })
 }
 
+const DELAY = 10000
+
+function useQueryTimeout({
+  queryKey,
+  errorKey,
+  cancel,
+  trigger,
+  removeTimeoutErrorAfter = DELAY,
+}: {
+  queryKey: string[]
+  errorKey: ErrorKey
+  trigger?: string
+  removeTimeoutErrorAfter?: number
+  cancel?: () => Promise<void> | void
+}): {
+  onError: (error: ErrorIDE) => void
+  timeout: (delay?: number) => void
+  onSuccess: () => void
+} {
+  const queryClient = useQueryClient()
+  const { addError, removeError } = useIDE()
+
+  let timeoutId: ReturnType<typeof setTimeout>
+
+  function timeout(delay: number = DELAY): void {
+    timeoutId = setTimeout(() => {
+      addError(errorKey, {
+        key: errorKey,
+        message: `Request timed out`,
+        description: `Request ${queryKey.join(
+          '->',
+        )} timed out after ${delay}ms`,
+        status: 408,
```

Comment:
Curious how this is used.

Reply:
Just pushing it to error stack and displaying in Errors, because it can indicate that there are some errors on the server

Reply:
But this is the client timing out, right? It's not the actual server HTTP status code.